Tests/NIOOpenSSLTests/ByteBufferBIOTest.swift

```diff
@@ -0,0 +1,328 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import XCTest
+import NIO
+import CNIOOpenSSL
+@testable import NIOOpenSSL
+
+
+final class ByteBufferBIOTest: XCTestCase {
+    override func setUp() {
+        guard openSSLIsInitialized else {
+            fatalError("Cannot run tests without OpenSSL")
+        }
+    }
+
+    private func bioWithDroppedRef() -> UnsafeMutablePointer<BIO> {
```

Comment:
this seems very unsafe, nothing is keeping `swiftBIO` alive

Reply:
That's the purpose of the function: to demonstrate that this works ok.